MusicApp.SongService.Application/CQRS/Commands/CreateSong/CreateSongCommandHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿using FluentValidation;
+using MediatR;
+using MusicApp.SongService.Application.Repositories;
+using MusicApp.SongService.Domain.Entities;
+
+namespace MusicApp.SongService.Application.CQRS.Commands.CreateSong;
+
+public class CreateSongCommandHandler : IRequestHandler<CreateSongCommand, Song>
+{
+    private readonly ISongRepository _repository;
+    private readonly CreateSongCommandValidator _validator;
+    
+    public CreateSongCommandHandler(ISongRepository repository)
+    {
+        _repository = repository;
+        _validator = new();
+    }
+
+    public async Task<Song> Handle(CreateSongCommand request, CancellationToken cancellationToken)
+    {
+        await _validator.ValidateAndThrowAsync(request, cancellationToken);
```

Comment:
you should rather add validation to a MediatR behavior pipeline to avoid manual validation or you can use automatic validation functionality provided by FluentValidation library, but it's not recommended for reasons which listed in FluentValidation documentation (second one is untruth)